internal/crypto/ptls/fips_strict.go

```diff
@@ -40,40 +39,32 @@ func init() {
 
 func Default(rootCAs *x509.CertPool) *tls.Config {
 	return &tls.Config{
-		// goboring requires TLS 1.2 and only TLS 1.2
-		MinVersion: SecureTLSConfigMinTLSVersion,
+		MinVersion: tls.VersionTLS12,
+		// goboring now also supports TLS 1.3 (see https://github.com/golang/go/issues/64717)
+		// so this default configuration can allow either 1.2 or 1.3
 		MaxVersion: SecureTLSConfigMinTLSVersion,
 
-		// enable HTTP2 for go's 1.7 HTTP Server
-		// setting this explicitly is only required in very specific circumstances
-		// it is simpler to just set it here than to try and determine if we need to
-		NextProtos: []string{"h2", "http/1.1"},
-
-		// optional root CAs, nil means use the host's root CA set
-		RootCAs: rootCAs,
-
-		// This is all of the fips-approved ciphers.
+		// This is all the fips-approved TLS 1.2 ciphers.
 		// The list is hard-coded for convenience of testing.
-		// This is kept in sync with the boring crypto compiler via TestFIPSCipherSuites.
+		// If this list does not match the boring crypto compiler's list then the TestFIPSCipherSuites integration
+		// test should fail, which indicates that this list needs to be updated.
```

Comment:
We tweaked this list after finding a mistake.  The next CI run should give us good insight.